### DISCUSSION TITLE
When can we finally use the cuda intellisense?

### DISCUSSION THREAD

**Discussion by marsmiao** [microsoft/vscode-cpptools#7520] [Q&A]
Hello there, I'm a fan of vscode-cpptools and I'm really excited about the newly coming feature, the cuda grammer highlight and so forth. But when I download the code-insiders, I cannot find how can I use this new feature, can you give me some tips on how to use cuda intellisense?

**Comment by sean-mcmanus** [comment]
I believe you need to install the CUDA SDK, set your compilerPath to the nvcc.exe, and open a .cu file. Let us know if that doesn't work for you.

**Reply by marsmiao** [reply]
Thanks. My environment is macos vscode, while I mainly use it to ssh remote linux server. On that server, I have installed CUDA 11 and set compilerPath to nvcc. However, when I open a .cu file and set the language mode to CUDA C++, there has no typical intellisense working. Also, I wonder if I have a project which use both gcc and nvcc, how can I set compilerPath? And there is a strange thing that when I open a .cu and change language mode to C++, it somehow works, but only a little bit; The include path or even datatype will be marked as wrong then

**Reply by sean-mcmanus** [reply]
@Colengms Do you know what could be happening?

**Reply by sean-mcmanus** [reply]
@marsmiao Can you set C_Cpp.loggingLevel to "Debug" and check your "C/C++" logging for "Invoking nvcc with command line:" and/or look in the logging another failure? The output of running C/C++: Log Diagnostics could also help diagnose what is going wrong. My guess is there's some failure to query nvcc.exe, perhaps due to some argument passed to it.

I think you usually should only need to set the nvcc.exe and it should work with C++ files too.

**Reply by Colengms** [reply]
Unfortunately, the NVIDIA CUDA Toolkit is no longer supported on macOS.  As per NVIDIA's direction, CUDA is not supported by the C/C++ extension on macOS.

https://developer.nvidia.com/nvidia-cuda-toolkit-developer-tools-mac-hosts
